### ISSUE TITLE
VSCode extension may be locking .dme file

### ISSUE THREAD

**Issue by DamianX** [SpaceManiac/SpacemanDMM#16]
Not the first time I encounter this, but I'm not sure how to reliably reproduce it.
Basically, I think the extension is locking the .dme so git can't write to it.

```
$ git pull pomf Bleeding-Edge
vgstation13.dme: Permission denied
```

Closing VS Code or disabling the extension fixes the problem.

I can try getting a test case if this is not enough info.

**Comment by SpaceManiac** [comment]
Antur brought this up before. This is due to the fact that the parser works line-by-line. The `.dme` is effectively locked until the parse finishes. This can be fixed by pre-buffering the entire file, but it may cause an inaccurate parse owing to the difference between the buffered file and the one you just checked out.

A workaround in the meantime is to pull before opening VS Code.

**Comment by SpaceManiac** [comment]
Fixed in 9c6cdd8a7b6dc3788b9ddbaa8ccdb3e2627d70ed.
